### PR TITLE
Accelerate BioCLIP training via `torchcompile`

### PR DESCRIPTION
**Pull request by vimar-gu**
The updates include:
* Adapt `open_clip` to 3.1.0
* Add `torchcompile` to the training script
* Update requirements for the training environment

The new training script finishes TOL-1M training in 6 hours (4 H100s) and TOL-10M training in 30 hours (8 H100s). Previously, we needed ~38h to finish the TOL-10M training. 

There is still one issue that, when submitted through Slurm, the model evaluation will lead to worker collapse. I skip the model evaluation for now, as it does not really affect the model training. But if people still want to have an eye on the validation set, they can run the training using the interactive mode. (I'm not sure why the issue only occurs in the batch job mode, and I will look into it later.)

### CODE REVIEW

**Comment by egrace479** [comment]
@thompsonmj or @NetZissou, any thoughts on the issue?

**Comment by egrace479** [comment]
> I have updated the evaluation code, including compatibility with the new `open_clip` version and less code repetition. We can merge it to the main branch first, as it saves a significant amount of time.

Sounds good! Should the requirements be updated then as well? Also, there was no code pulled directly from them, right (want to make sure our `HISTORY.md` stays up to date)?

**Comment by vimar-gu** [comment]
We upgraded `open_clip` to 3.1.0, which has been noted in the commits. 

I have also updated some requirements.

**Review by egrace479 — APPROVED** [review]
